intro-HPC/ch_openFOAM.tex

```diff
@@ -0,0 +1,319 @@
+      or using command-line tools like \lstinline|postProcess|;
+      see also \url{https://cfd.direct/openfoam/user-guide/postprocessing}.
+\end{itemize}
+
+Depending on the size of the domain and the desired format of the results, these pre- and post-processing
+steps can be run either before/after the job running the actual simulation, either on the HPC infrastructure
+or elsewhere, or as a part of the job that runs the OpenFOAM simulation itself.
+
+Do make sure you are using the same OpenFOAM version in each of the steps.
+Meshing can be done sequentially (i.e., on a single core) using for example \lstinline|blockMesh|,
+or in parallel using more advanced meshing tools like \lstinline|snappyHexMesh|, which is highly recommended
+for large cases. For more details, see \url{https://cfd.direct/openfoam/user-guide/mesh/}.
+
+One important aspect to keep in mind for `offline' pre-processing is that the domain decomposition needs to match
+the number of processor cores that are used for the actual simulation,
+see also \autoref{sec:best-practices-openfoam-domain-decomposition-processor-cores}.
+
+For post-processing you can either download the simulation results to a local workstation,
+or do the post-processing (interactively) on the HPC infrastructure, for example on the login nodes
+or using an interactive session on a workernode. This may be interesting to avoid the overhead of
+downloading the results locally.
+
+\section{Running OpenFOAM in parallel}
+
+For general information on running OpenFOAM in parallel,
+see \url{https://cfd.direct/openfoam/user-guide/running-applications-parallel/}.
+
+\subsection{The \texttt{-parallel} option}
+
+When running OpenFOAM in parallel, \strong{do not forget to specify the \texttt{\small{-parallel}} option},
+to avoid running the same OpenFOAM simulation $N$ times, rather than running it once using $N$ processor cores.
+
+You can check whether OpenFOAM was run in parallel in the output of the main command:
+the OpenFOAM header text should only be included \emph{once} in the output, and it should specify a value different
+than `\lstinline|1|' in the \lstinline|nProcs| field. Note that most pre- and post-processing utilities like
+\lstinline|blockMesh|, \lstinline|decomposePar| and \lstinline|reconstructPar| can not be run in parallel.
+
+\subsection{Using \texttt{mympirun}}
+
+It is highly recommended to use the \lstinline|mympirun| command when running parallel OpenFOAM simulations
+rather than the standard \lstinline|mpirun| command;
+see \autoref{sec:best-practices-mympirun} for more information on \lstinline|mympirun|.
+
+To use \lstinline|mympirun|, make sure that the \lstinline|vsc-mympirun| module is loaded.
+
+\begin{prompt}
+%\shellcmd{module load vsc-mympirun}%
+\end{prompt}
+
+Note that you should \emph{not} specify a specific version here,
+see also \autoref{sec:best-practices-mympirun-module}.
```

Comment:
Do we explain in the `mympirun` chapter why not to specify a version here? If not, please add.

The reason is to ensure that the latest version of the `mympirun` script is always used, since it may include important bug fixes or improvements.

Reply:
Now we do :)